src/collectors/processcheck/processcheck.py

```diff
@@ -0,0 +1,183 @@
+# coding=utf-8
+
+"""
+A Diamond collector that checks process defined in it's
+config file by matching them with their executable filepath or the process name.
+
+Example config file ProcessCheckCollector.conf
+
+```
+enabled=True
+ttl=150
+[process]
+[[postgres]]
+exe=^\/usr\/lib\/postgresql\/+d.+d\/bin\/postgres$
+name=^postgres,^pg
+
+```
+exe and name are both lists of comma-separated regexps.
+
+"""
+
+import re
+
+import diamond.collector
+import diamond.convertor
+
+try:
+    import psutil
+    psutil
+except ImportError:
+    psutil = None
+
+try:
+    import netuitive
+except ImportError:
+    netuitive = None
+
+
+def match_process(pid, name, cmdline, exe, cfg):
+    """
+    Decides whether a process matches with a given process descriptor
+
+    :param pid: process pid
+    :param exe: process executable
+    :param name: process name
+    :param cmdline: process cmdline
+    :param cfg: the dictionary from processes that describes with the
+        process group we're testing for
+    :return: True if it matches
+    :rtype: bool
+    """
+    for exe_re in cfg['exe']:
+        if exe_re.search(exe):
+            return True
+    for name_re in cfg['name']:
+        if name_re.search(name):
+            return True
+    for cmdline_re in cfg['cmdline']:
+        if cmdline_re.search(' '.join(cmdline)):
+            return True
+    return False
+
+
+def get_value(process, name):
+    result = getattr(process, name)
+    try:
+        return result()
+    except TypeError:
+        return result
+
+
+class ProcessCheckCollector(diamond.collector.Collector):
+
+    def __init__(self, *args, **kwargs):
+        super(ProcessCheckCollector, self).__init__(*args, **kwargs)
+
+        self.hostname = self.get_hostname()
+        self.ttl = self.config['ttl']
+
+        if not netuitive:
+            self.log.error('netuitive import failed. ProcessCheckCollector disabled')
+            self.enabled = False
+            return
+
+        try:
+            self.version = self._get_version()
+            self.api = netuitive.Client(self.config['url'], self.config['api_key'], self.version)
+        except Exception as e:
+            self.log.debug(e)
+
+
+    def process_config(self):
+        super(ProcessCheckCollector, self).process_config()
+        """
+        prepare self.processes, which is a descriptor dictionary in
+        pg_name: {
+            exe: [regex],
+            name: [regex],
+            cmdline: [regex],
+            procs: [psutil.Process]
+        }
+        """
+        self.processes = {}
+        self.processes_info = {}
+        for pg_name, cfg in self.config['process'].items():
+            pg_cfg = {}
```

Comment:
What does `pg` stand for?

Reply:
I'd think it's program

Reply:
Could you refactor to name it program so it's clear? More descriptive variable names are free.